### DISCUSSION TITLE
Corporate usage with Entra ID Passwordless authentication

### DISCUSSION THREAD

**Discussion by ArakelYorghanjyan** [canonical/authd#595] [Feature Requests]
Hi, guys!
We want to implement "autd" in our corporate environment, but the first problem is the issue of logging into the system of a user who has the [Passwordless ](https://learn.microsoft.com/en-us/entra/identity/authentication/concept-authentication-passwordless) mode enabled in Entra ID.
The question is: whether in this mode it will be possible to create a PIN or a local password that policies can change, say, once every three months. This is necessary since the user initially does not have a password.
On a Windows PC, this problem is overcome using Web sign-in and the initial setup of Windows Hello,
i.e. the user can use a PIN, fingerprint, and Face ID to log in.
Is it possible to have a similar solution using "autd"?

**Comment by adombeck** [comment]
Hi, thanks for reaching out!

I'm not sure I understand your question. When a user logs in via authd, they are asked to set a local password, which they can use to authenticate in subsequent login attempts. Does that solve your use case?

Regarding password expiration policies, we currently don't plan to support such policies for the local password. Note that password expiration policies are usually not considered beneficial for security (unless there is evidence of a compromise) and harmful for user experience (see for example [this article](https://arstechnica.com/information-technology/2016/08/frequent-password-changes-are-the-enemy-of-security-ftc-technologist-says/?imm_mid=0e6947&cmp=em-webops-na-na-newsltr_security_20160809)).

**Reply by ArakelYorghanjyan** [reply]
If the user login via authd, he will be prompted to enter the network and local password.
This solution is perfect with one BUT: Since the "Passwordless" scenario will be used on the Entra ID side, the user can forget his local password.
Will the user be able to click on the "Forgot password" button, re-authenticate via the QR code, and set a new local password?

**Reply by 3v1n0** [reply]
Once the authentication mode has been pre-selected, one can always go back (with `r` in the ssh case or via `Escape` key otherwise) and choose the device-based authentication method, isn't it?

**Reply by ArakelYorghanjyan** [reply]
I felt misunderstood.
Since the corporate account has no password, the user will be forced to use the local password of the device in any case.
My question is, is there a mechanism for resetting this local password on the device if the user forgets it?

**Reply by adombeck** [reply]
> My question is, is there a mechanism for resetting this local password on the device if the user forgets it?

Yes, when you log in via SSH, you can set a new local password by entering `r` in the password prompt, then selecting the authentication method "Device Authentication" and completing the authentication. You will then be asked to set a new local password.

**Reply by ArakelYorghanjyan** [reply]
And what if, the user sitting in front of the device and trying to sign in , and of course the SSH is not configured on that device?
I think we need a button, that will activate the password reset function?

**Reply by adombeck** [reply]
You can also already reset the password in GDM (GNOME's display/login manager). When you see the "Enter your local password" prompt, click on the back button and choose "Device Authentication". Same as with SSH, you will be asked to set a new local password after successful authentication.

**Reply by ArakelYorghanjyan** [reply]
That's great solution!
Thanks!

**Reply by 3v1n0** [reply]
In GDM / lockscreen views you can switch authentication method also from the bottom-right "key" icon.

**Comment by ArakelYorghanjyan** [comment]
#709 and #731 can be part of this discussion, as I understand the TPM implementation will remedy our three problems.
Is there a place for the TPM implementation discussion?

**Comment by nooreldeenmansour** [comment]
Hi @ArakelYorghanjyan 

We recently released support for Passwordless flow and FIDO2 keys for the Entra ID broker. It is available at the edge channels. 

We would appreciate if you test it and report back your findings/experience :)

**Reply by gbinetti** [reply]
We are introducing authd in our company, but also device auth flow will be soon replaced for all accounts by passwordless/FIDO2 flows. Do you have an idea how long it takes for these changes to make it to the stable channel?

**Reply by nooreldeenmansour** [reply]
Hi @gbinetti 

We are planning to do a stable release within a month.

 In the meantime, you can test and experiment with the flow from the edge PPA/broker channels, and let us know if you have any feedback

**Reply by gbinetti** [reply]
Thank you for the quick response!

**Reply by gbinetti** [reply]
@nooreldeenmansour I'm testing the edge channel currently and all seems working as intended. One thing I'd like to give the feedback on is that the ssh authentication always defaults to Local Password, but I think there should be a way to either have the menu for selecting the preferred authentication at the beginning of the auth flow or enforce the default option in the broker conf in same way something like "default_auth_flow = EntraID". Is there already any plan for that?

**Reply by nooreldeenmansour** [reply]
hi @gbinetti 

Currently, the default authentication method for an already logged-in account is the local password. You can press `r` to go back and select Entra ID auth flow though.

It's already planned as part of the upcoming cycle to add an option to disable local password via broker config

**Comment by nooreldeenmansour** [comment]
The new Entra ID sign-in flows which support passwordless sign-ins for eligible users have finally been released to the stable PPA and will soon be available in the archive

You can install `0.5.0` from the stable channel for the the authd-msentraid snap broker, and authd `0.7.0` from the [stable PPA](https://launchpad.net/~ubuntu-enterprise-desktop/+archive/ubuntu/authd)
